includes/abstracts/class-wc-gateway-mondido-abstract.php

```diff
@@ -226,87 +86,22 @@ public function add_order_fee($fee, &$order, $qty = 1) {
 		}
 	}
 
-	/**
-	 * @param $merchant_id
-	 * @param $password
-	 * @param $transaction_id
-	 *
-	 * @return array|mixed|object
-	 */
-	public static function __lookup( $merchant_id, $password, $transaction_id ) {
-		$result = wp_remote_get( 'https://api.mondido.com/v1/transactions/' . $transaction_id, array(
-			'headers' => array(
-				'Authorization' => 'Basic ' . base64_encode( "{$merchant_id}:{$password}" )
-			)
-		) );
-
-		if ( is_a( $result, 'WP_Error' ) ) {
-			throw new Exception( implode( $result->errors['http_request_failed'] ) );
-		}
-
-		if ( $result['response']['code'] != 200 ) {
-			throw new Exception( $result['body'] );
-		} else {
-			return json_decode( $result['body'], TRUE );
-		}
-	}
-
-	/**
-	 * Lookup transaction data
-	 *
-	 * @param $transaction_id
-	 *
-	 * @return array|bool
-	 */
 	public function lookupTransaction( $transaction_id ) {
-		try {
-			return self::__lookup( $this->merchant_id, $this->password, $transaction_id );
-		} catch ( Exception $e ) {
-			if ( strpos( $e->getMessage(), 'errors.transaction.not_found' ) !== FALSE ) {
-				// Workaround for errors.transaction.not_found
-				$attempt = 0;
-				do {
-					sleep( 5 );
-					$this->log( "lookupTransaction (not found error). Transaction ID: {$transaction_id}. Error: {$e->getMessage()}. Attempt: {$attempt}", WC_Log_Levels::WARNING );
-
-					try {
-						return self::__lookup( $this->merchant_id, $this->password, $transaction_id );
-					} catch ( Exception $e ) {
-						$attempt++;
-						if ( $attempt > 12 ) {
-							// wc_add_notice( $e->getMessage(), 'error' );
-							break;
-						}
-					}
-				} while ( TRUE );
-			}
-
-			wc_add_notice( $e->getMessage(), 'error' );
+		$transaction = $this->transaction->get($transaction_id);
+		if (is_wp_error($transaction)) {
+			wc_add_notice( $transaction->get_error_message(), 'error' );
+			return false;
 		}
-
-		return FALSE;
+		return json_decode(json_encode($transaction), true);
 	}
 
 	public function lookupTransactionByOrderId( $order_id ) {
-		$result = wp_remote_get("https://api.mondido.com/v1/transactions?filter[payment_ref]=$order_id", array(
-			'headers' => array(
-				'Authorization' => 'Basic ' . base64_encode( "{$this->merchant_id}:{$this->password}" )
-			)
-		) );
-
-		if ( is_a( $result, 'WP_Error' ) ) {
-			throw new Exception( implode( $result->errors['http_request_failed'] ) );
-		}
-
-		if ( $result['response']['code'] != 200 ) {
-			throw new Exception( $result['body'] );
-		} else {
-			$data = json_decode( $result['body'], TRUE );
-			if (count($data) === 0) {
-				return null;
-			}
-			return $data[0];
+		$transaction = $this->transaction->get_by_reference($order_id);
+		if (is_wp_error($transaction)) {
+			wc_add_notice( $transaction->get_error_message(), 'error' );
+			return false;
 		}
+		return json_decode(json_encode($transaction), true);
```

Comment:
Why encode and decode?

Reply:
to convert object to array ;-)

Reply:
the old api call decoded everything to an array, the new one is decoding to object, so to not break old code i just replaced the inner of the old functions with the new api and reencoded the result to array, to keep the old behaviour.
new code is using the transaction property directly